File: src/ros_essentials_cpp/src/assigment_2/rectangle_area_server.cpp
#include "ros/ros.h"
#include "ros_essentials_cpp/RectangleArea.h"

bool calculate_area(ros_essentials_cpp::RectangleArea::Request &req,
                    ros_essentials_cpp::RectangleArea::Response &res) {
    try {
        res.area = req.width * req.height;
        ROS_INFO("request: width=%f, height=%f", req.width, req.height);
        ROS_INFO("sending back response: [%f]", res.area);
        return true;
    } catch (...) {
        ROS_ERROR("ERROR OCURRES");
        return false;
    }
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "rectangle_area_server");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("rectangle_area_service", calculate_area);
    ROS_INFO("Ready to calculate a rectangle area.");
    ros::spin();

    return 0;
}

File: src/ros_essentials_cpp/src/assigment_2/scripts/rectangle_area_server.py
#!/usr/bin/env python

import rospy
from ros_essentials_cpp.srv import RectangleArea, RectangleAreaRequest, RectangleAreaResponse

def handle_rectangle_area(req):
    # type: (RectangleAreaRequest) -> RectangleAreaResponse
    """
    Handles the server request and returns the area of the rectangle as a response
    """
    print "Returning [%s * %s = %s]"%(req.width, req.height, (req.width * req.height))
    return RectangleAreaResponse(req.width * req.height)

def rectangle_area_server():
    # type: () -> None
    """
    Initializes the server and spins until Ctrl+C
    """
    rospy.init_node('rectangle_area_server')
    s = rospy.Service('rectangle_area_service', RectangleArea, handle_rectangle_area)
    print "Ready to calculate a rectangle area"
    s.spin()
    
if __name__ == "__main__":
    try:
        rectangle_area_server()
    except:
        pass